src/khoj/processor/conversation/openai/utils.py

```diff
@@ -112,24 +123,35 @@ def llm_thread(g, messages, model_name, temperature, openai_api_key=None, api_ba
             client: openai.OpenAI = openai_clients[client_key]
 
         formatted_messages = [{"role": message.role, "content": message.content} for message in messages]
+        stream = True
+
+        # Update request parameters for compatability with o1 model series
+        if model_name.startswith("o1"):
+            stream = False
+            temperature = 1
+            model_kwargs.pop("stop", None)
+            model_kwargs.pop("response_format", None)
```

Comment:
Refer to documentation on these parameter decisions here. Let's keep an eye on it as the beta rolls out: https://platform.openai.com/docs/guides/reasoning/beta-limitations

Reply:
Maybe add a link to the doc in the code? Will make it easier to remove the custom code later on hopefully.